src/components/spreadsheet/equipment-table.js

```diff
@@ -29,8 +35,74 @@ export const EquipmentTable = ({
     fetched,
     network,
     shouldHidePinnedHeaderRightBorder,
+    onCellClicked,
 }) => {
     const theme = useTheme();
+    const { snackError } = useSnackMessage();
+    const [clickedCellData, setClickedCellData] = React.useState({});
+    const [generatorId, setGeneratorId] = React.useState('');
+    const [
+        popupGeneratorEditRegulatingTerminalFields,
```

Comment:
popupGeneratorEditRegulatingTerminalFields is too long as name and I don't think we need all this details.
We only have one dialogue so you can use something like this : **openDialog**